src/query/case.rs

```diff
@@ -0,0 +1,125 @@
+use crate::{Condition, Expr, IntoCondition, SimpleExpr};
+
+#[derive(Debug, Clone)]
+pub struct CaseStatementCondition {
+    pub(crate) condition: Condition,
+    pub(crate) result: Expr,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct CaseStatement {
+    pub(crate) when: Vec<CaseStatementCondition>,
+    pub(crate) r#else: Option<Expr>,
+}
+
+impl CaseStatement {
+    /// Creates a new case statement expression
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::{*, tests_cfg::*};
+    ///
+    /// let query = Query::select()
+    ///     .expr_as(
+    ///         CaseStatement::new()
+    ///             .case((Expr::tbl(Glyph::Table, Glyph::Aspect).is_in(vec![2, 4]), Expr::val(true)))
+    ///             .finally(Expr::val(false)),
+    ///          Alias::new("is_even")
+    ///     )
+    ///     .from(Glyph::Table)
+    ///     .to_owned();
+    ///
+    /// assert_eq!(
+    ///     query.to_string(PostgresQueryBuilder),
+    ///     r#"SELECT (CASE WHEN ("glyph"."aspect" IN (2, 4)) THEN TRUE ELSE FALSE END) AS "is_even" FROM "glyph""#
+    /// );    
+    /// ```
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Adds new `CASE WHEN` to existing case statement.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::{*, tests_cfg::*};
+    ///
+    /// let query = Query::select()
+    ///     .expr_as(
+    ///             Expr::case((
+    ///                 Expr::tbl(Glyph::Table, Glyph::Aspect).gt(0),
```

Comment:
Finally, I'd appreciate at least one example is a complex condition, may be `Cond::all().add(...)` while the other is a `SimpleExpr` to illustrate the API